pkg/checks/traceroute/traceroute.go

```diff
@@ -126,44 +176,69 @@ func TraceRoute(ctx context.Context, cfg tracerouteConfig) (map[int][]Hop, error
 	var wg sync.WaitGroup
 
 	for ttl := 1; ttl <= cfg.MaxHops; ttl++ {
+		c, hopSpan := tr.Start(ctx, fmt.Sprintf("Hop %d", ttl), trace.WithAttributes(
+			attribute.Int("ttl", ttl),
+		))
 		wg.Add(1)
 		go func(ttl int) {
 			defer wg.Done()
+			defer hopSpan.End()
+
 			l := log.With("ttl", ttl)
-			logctx := logger.IntoContext(ctx, l)
+			logctx := logger.IntoContext(c, l)
+
+			retry := 0
 			err := helper.Retry(func(ctx context.Context) error {
```

Comment:
`err` redeclaration here - I think we can just assign.

Reply:
Weird that the linter doesn't complain about it but thanks 🙂